Exiled.API/Structs/AttachmentIdentifier.cs

```diff
@@ -0,0 +1,248 @@
+// -----------------------------------------------------------------------
+// <copyright file="AttachmentIdentifier.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.API.Structs
+{
+    using System;
+    using System.Linq;
+
+    using Exiled.API.Enums;
+
+    using InventorySystem.Items.Firearms.Attachments;
+
+    /// <summary>
+    /// A tool to identify attachments.
+    /// </summary>
+    public struct AttachmentIdentifier
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AttachmentIdentifier"/> struct.
+        /// </summary>
+        /// <param name="name"><inheritdoc cref="Name"/></param>
+        public AttachmentIdentifier(AttachmentNameTranslation name)
+        {
+            Code = 0;
+            Name = name;
+            Slot = default;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AttachmentIdentifier"/> struct.
+        /// </summary>
+        /// <param name="name"><inheritdoc cref="Name"/></param>
+        /// <param name="slot"><inheritdoc cref="Slot"/></param>
+        public AttachmentIdentifier(AttachmentNameTranslation name, AttachmentSlot slot)
+        {
+            Code = 0;
+            Name = name;
+            Slot = slot;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AttachmentIdentifier"/> struct.
+        /// </summary>
+        /// <param name="code"><inheritdoc cref="Code"/></param>
+        /// <param name="name"><inheritdoc cref="Name"/></param>
+        /// <param name="slot"><inheritdoc cref="Slot"/></param>
+        internal AttachmentIdentifier(uint code, AttachmentNameTranslation name, AttachmentSlot slot)
+        {
+            Code = code;
+            Name = name;
+            Slot = slot;
+        }
+
+        /// <summary>
+        /// Gets the attachment code.
+        /// </summary>
+        public uint Code { get; }
+
+        /// <summary>
+        /// Gets the attachment name.
+        /// </summary>
+        public AttachmentNameTranslation Name { get; }
+
+        /// <summary>
+        /// Gets the attachment slot.
+        /// </summary>
+        public AttachmentSlot Slot { get; }
+
+        /// <summary>
+        /// Compares two operands: <see cref="AttachmentIdentifier"/> and <see cref="AttachmentIdentifier"/>.
+        /// </summary>
+        /// <param name="left">The left-hand <see cref="AttachmentIdentifier"/> operand to compare.</param>
+        /// <param name="right">The right-hand <see cref="AttachmentIdentifier"/> operand to compare.</param>
+        /// <returns><see langword="true"/> if the values are equal.</returns>
+        public static bool operator ==(AttachmentIdentifier left, AttachmentIdentifier right) => left == right;
```

Comment:
What does this do? It kind of looks like a stackoverflow to me.

Reply:
I don't think? Anyway I'm gonna change it rn.